service/matching/ack_manager_test.go

```diff
@@ -28,35 +28,49 @@ import (
 	"math/rand"
 	"testing"
 
+	"github.com/golang/mock/gomock"
+
 	"github.com/stretchr/testify/require"
-	"go.temporal.io/server/common/log"
 )
 
 func TestAckManager_AddingTasksIncreasesBacklogCounter(t *testing.T) {
+	controller := gomock.NewController(t)
+	defer controller.Finish()
```

Comment:
for the tests where you do call NewController (not in suites), you don't need to call controller.Finish. the controller registers itself with t.Cleanup so it happens automatically

Reply:
I shall clean this up here - however, I do see them being used extensively in other places all over the repo too. I shall remember to clean up this. Thanks!